Source/Shared/arcana/threading/internal/internal_task.h

```diff
@@ -338,7 +347,20 @@ namespace arcana
         template<typename ResultT, typename ErrorT, typename CallableT>
         std::shared_ptr<task_payload_with_return<ResultT, ErrorT>> make_work_payload(CallableT&& callable)
         {
-            return std::make_shared<task_payload_with_work<ResultT, ErrorT, sizeof(callable)>>(std::forward<CallableT>(callable));
+            using PayloadT = task_payload_with_work<ResultT, ErrorT, sizeof(callable)>;
+            struct DeleterT
+            {
+                void operator()(PayloadT* ptr)
+                {
+                    if (auto error = ptr->get_unhandled_error())
+                    {
+                        if constexpr (std::is_same<std::exception_ptr, ErrorT>::value)
+                            std::rethrow_exception(*error);
+                    }
+                    delete ptr;
+                }
+            };
```

Comment:
nit: maybe a [lambda deleter](https://stackoverflow.com/questions/10319009/unique-ptrt-lambda-custom-deleter-for-array-specialization) will be better?